### DISCUSSION TITLE
Facing ImagePullBackOff issue after injecting Linkerd.

### DISCUSSION THREAD

**Discussion by nm158b** [linkerd/linkerd2#6690] [Q&A]
I have Installed **linkerd** and **linkerd-viz** on Azure AKS. 
When injecting linkerd to my namespace, using the following command.
`kubectl get -n <namespace> deploy -o yaml | linkerd inject - | kubectl apply -f -`

But some of the pods keep getting restarted and below is the error message I see in the events and logs. These worked fine before injecting linkerd.

ERROR: ImagePullBackOff

**Pods Status:**
```
> kubectl get pods 
NAME                             READY   STATUS             RESTARTS   AGE

pb-titan-gt3-6c69b48947-t5cw4    1/1     Running            0          19d
pb-titan-gt3-895bcd8b-lg8mw      1/2     ErrImagePull       0          5d4h
pb-titan-sgi-5c4f94f7b-22s4q     1/2     ImagePullBackOff   0          6d21h
pb-titan-sgi-86c44846cf-2sxln    1/1     Running            0          3h16m
pb-titan-so-66fcf9d74-k94tb      1/2     ImagePullBackOff   0          4d3h
```

**Deployment Events:**
```
Events:
  Type     Reason   Age                        From     Message
  ----     ------   ----                       ----     -------
  Normal   BackOff  34m (x44062 over 6d22h)    kubelet  Back-off pulling image "attpandoraboxacr.azurecr.io/com.att.pandorasbox/pb-titan-sgi-prod:1.0.1"
  Warning  Failed   4m25s (x44194 over 6d22h)  kubelet  Error: ImagePullBackOff
```
Any help is appreciated, Thank you.

**Comment by kleimkuhler** [comment]
Is there any chance these issues existed before injecting Linkerd? I haven't thought of a reason that after editing these pod specs to include an additional container/image (the `linkerd-proxy`), that it would lead to image pull issues for the _other_ container. If you can double check that this is indeed connected to Linkerd being injected that would be helpful!
